src/Microsoft.Health.Fhir.SqlServer/Features/Search/Expressions/Visitors/QueryGenerators/SqlQueryGenerator.cs

```diff
@@ -392,19 +441,47 @@ private void HandleTableKindNormal(SearchParamTableExpression searchParamTableEx
 
         private void HandleTableKindAll(SearchParamTableExpression searchParamTableExpression)
         {
-            StringBuilder.Append("SELECT ")
-                .Append(VLatest.Resource.ResourceTypeId, null).Append(" AS T1, ")
-                .Append(VLatest.Resource.ResourceSurrogateId, null).AppendLine(" AS Sid1")
-                .Append("FROM ").AppendLine(VLatest.Resource);
+            int predecessorIndex = FindRestrictingPredecessorTableExpressionIndex();
 
-            using (var delimited = StringBuilder.BeginDelimitedWhereClause())
+            if (_unionAllVisited && predecessorIndex > 0 && searchParamTableExpression.ChainLevel == 0)
```

Comment:
If you have some insight to what this function does, please add some comments :) method level would be ok too